libs/product/driver/magento/src/transforms/product-transformers.ts

```diff
@@ -8,27 +8,48 @@ import {
 } from '../models/magento-product';
 import { transformMagentoBundledProduct } from './bundled-product-transformers';
 import { transformMagentoConfigurableProduct } from './configurable-product-transformers';
+import { transformMagentoProductPreview } from './product-preview';
 import { transformMagentoSimpleProduct } from './simple-product-transformers';
 
+/**
+ * Transforms a Magento product into a Daffodil product.
+ * Handles all product types.
+ */
+export function transformProduct(product: MagentoProduct, mediaUrl: string): DaffProduct {
+  return transformMagentoProduct(transformMagentoSimpleProduct(product, mediaUrl), product);
+}
+
+/**
+ * Transforms a Magento product preview into a minimal Daffodil product.
+ * Handles all product types.
+ */
+export function transformProductPreview(product: MagentoProduct, mediaUrl: string): DaffProduct {
```

Comment:
I think the names of `transformProduct` and `transformProductPreview` are a little confusing and I don't think they are necessary. I think it should just be the `transformMagentoProduct` function here:
```
export function transformMagentoProduct(product: MagentoProduct): DaffProduct {
  switch(product.__typename) {
    case MagentoProductTypeEnum.BundledProduct:
      return transformMagentoBundledProduct(transformMagentoSimpleProduct(product), <MagentoBundledProduct>product);
    case MagentoProductTypeEnum.ConfigurableProduct:
      return transformMagentoConfigurableProduct(transformMagentoSimpleProduct(product), <MagentoConfigurableProduct>product);
    case MagentoProductTypeEnum.SimpleProduct:
    default:
      return transformMagentoSimpleProduct(product);
  }
}
```
and then a congruent function in previews. Otherwise the api here is pretty confusing

Reply:
changed